guides/plugins/plugins/framework/event/add-custom-event.md

```diff
@@ -0,0 +1,116 @@
+# Add custom event
+
+## Overview
+
+In this guide you'll learn how to create your own event. You can read more about events in the [Symfony documentation](https://symfony.com/doc/current/event_dispatcher.html). 
+
+## Prerequisites
+
+In order to create your own event for your plugin, you first need a plugin as base. 
+Therefore, you can refer to the [Plugin Base Guide](../../plugin-base-guide.md).
+
+## Event interfaces and classes
+
+In Shopware, you have multiple interfaces and classes for different types of events, in the following you can find a list of them:
+
+- `ShopwareEvent`: This interface is just a basic event providing a `Context`, we need for almost all events.
+- `ShopwareSalesChannelEvent`: This interface extends from `ShopwareEvent` and provides additionally a `SalesChannelContext`.
+- `SalesChannelAware`: This interface provides the `SalesChannelId`.
+- `GenericEvent`: This interface will be used if you want to give your event a specific name like the database events (e.g. `product.written.`). Otherwise, you have to reference to the event class.
+- `NestedEvent`: This class will be used for events using another events for example the `EntityDeletedEvent` extends from the `EntityWrittenEvent`.
+- `BusinessEventInterface`: This interface extends from `ShopwareEvent` and will be used for dynamically assignment and is always named.
+
+## Create the event class
+
+First, we create a new class for our event, which we name `ExampleEvent`. In this example we implement the `Shopware\Core\Framework\Event\ShopwareSalesChannelEvent`.
+As mentioned above our class already implements a method for the `SalesChannelContext` and the `Context`. 
+Now we inject an `ExampleEntity` and the `SalesChannelContext` through the constructor and create a function which returns our `ExampleEntity`.
```

Comment:
```suggestion
Now we pass an `ExampleEntity` and the `SalesChannelContext` through the constructor and create a function which returns our `ExampleEntity`.
```

There is no dependency injection involved here, but rather just normal instance creation